index.bs

```diff
@@ -181,96 +182,66 @@ partial interface Navigator {
-  // Methods
-  Promise&lt;void&gt; supportsSession(optional XRSessionCreationOptions options);
-  Promise&lt;XRSession&gt; requestSession(optional XRSessionCreationOptions options);
-};
-</pre>
-
-An {{XRDevice}} represents a physical unit of XR hardware that can present imagery to the user somehow. On desktop devices this may take the form of a headset peripheral; on mobile devices it may represent the device itself in conjunction with a viewer harness. It may also represent devices without the ability to present content in stereo but with advanced (6DoF) tracking capabilities.
-
-Each {{XRDevice}} has a <dfn>supports immersive</dfn> value, which is a boolean which MUST be set to <code>true</code> if the device can support [=immersive sessions=] and <code>false</code> if it cannot.
-
-Each {{XRDevice}} has an <dfn>active immersive session</dfn>, which MUST be initially <code>null</code>, and a <dfn>list of non-immersive sessions</dfn>, which MUST be initially empty.
+Each [=XR device=] has a <dfn>supports immersive</dfn> value, which is a boolean which MUST be set to <code>true</code> if the device can support [=immersive sessions=] and <code>false</code> if it cannot.
 
-<div class="algorithm" data-algorithm="supports-session">
+<div class="algorithm" data-algorithm="supports-session-mode">
 
-When the <dfn method for="XRDevice">supportsSession(|options|)</dfn> method is invoked, it MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
+When the <dfn method for="XR">supportsSession(|options|)</dfn> method is invoked, it MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
 
-  1. Let |device| be the target {{XRDevice}} object.
-  1. If the |options| are not [=supported by the device=] |device|, [=reject=] |promise| with <code>null</code>.
+  1. [=Ensure an XR device is selected=].
+  1. If [=XR device=] is <code>null</code>, [=reject=] |promise| with <code>null</code>.
+  1. If the |options| are not [=supported by the XR device=], [=reject=] |promise| with a {{NotSupportedError}} and abort these steps.
   1. Else [=/resolve=] |promise| with <code>null</code>.
 
 </div>
 
+Calling {{XR/supportsSession()}} MUST NOT trigger device-selection UI as this would cause many sites to display XR-specific dialogs early in the document lifecycle without user activation.
+
+The {{XR}} object has an <dfn>active immersive session</dfn>, which MUST be initially <code>null</code>, and a <dfn>list of non-immersive sessions</dfn>, which MUST be initially empty.
+
 <div class="algorithm" data-algorithm="request-session">
 
-When the <dfn method for="XRDevice">requestSession(|options|)</dfn> method is invoked, the user agent MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
+When the <dfn method for="XR">requestSession(|options|)</dfn> method is invoked, the user agent MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
```

Comment:
Also, "in parallel"?

Reply:
To clarify the "In parallel" text is a standard web spec thing that means "in parallel to the main thread". Good resource for this kind of minutia: https://garykac.github.io/procspec/ (Search for "in parallel" in that doc)